### DISCUSSION TITLE
Data encryption

### DISCUSSION THREAD

**Discussion by sh4hri4roo** [bmresearch/Solnet#398] [Q&A] [answered]
Hello, I'm trying to encrypt/decrypt private messages between 2 parties rather than sign/verify. I just want to know if Solnet supports that?

**Comment by hoakbuilds** [comment, marked as answer]
Technically Solnet does not support this, Ed25519 keys used by Solana are not technically secure for encryption and decryption as they are specifically made for EdDSA, the edwards curve digital signature algorithm. You can convert these keys to X25519 and thus perform encryption and decryption using those keys, but it is not a trivial thing to do. Here's a cool [read](https://words.filippo.io/using-ed25519-keys-for-encryption/) about it.

**Reply by sh4hri4roo** [reply]
Thank you for reply
